ops/framework.py

```diff
@@ -782,18 +782,12 @@ class SomeObject:
             RuntimeError: if bound_event or observer are the wrong type.
         """
         if not isinstance(bound_event, BoundEvent):
-            raise RuntimeError(
+            raise TypeError(
                 f'Framework.observe requires a BoundEvent as second parameter, got {bound_event}')
+        # Help users of older versions of the framework.
```

Comment:
Nit: I think we should just drop this comment now, as I think it's more confusing than not having it.

Reply:
Ah, yeah, I just completely missed removing that line - I didn't intend for it to stay.